### ISSUE TITLE
When launching WebAssembly hosted app from VS2022 Pre5, changes are not shown

### ISSUE THREAD

**Issue by SteveSandersonMS** [dotnet/aspnetcore#36225]
This is probably related to hot reload, but I don't know for sure.

* Use VS2022Pre5 to create a WebAssembly hosted application
* Ctrl+F5 to run it
* Try to edit `Pages\Index.razor` to change `<h1>Hello, world!</h1>` to `<h1>Hello, world2!</h1>`
* See it fail to apply the change (filed separately as https://github.com/dotnet/aspnetcore/issues/36218)
* Now explictly build everything with Ctrl+Shift+b, relaunch with Ctrl+F5, and when it comes up in the browser, do a hard-refresh to ensure you're really seeing the latest

Expected: Page title should be `Hello, world2!`

Actual: It's still `Hello, world!`

In other words, I don't know how to make changes show up any more.

**Comment by SteveSandersonMS** [comment]
I suspect this is the same as https://github.com/aspnet/AspNetCore-ManualTests/issues/893

From investigation, it seems that the client assembly (e.g., `YourApp.Client.dll`) does *not* get updated on disk when rebuilding. Using ILSpy I can see that it continues to contain the old code.

Also:

 * Even after deleting `obj` and `bin` and rebuilding on the command line, the emitted assembly continues to contain the old code.
 * Even after forcibly killing any background `VBCSCompiler` process and rebuilding on the command line, the emitted assembly continues to contain the old code.
 * **But** if I do both of the above (kill `VBCSCompiler` *and* delete `obj` and `bin` then rebuild), the emitted assembly finally gets the new code

I don't completely know how this can be the case. Does the source generator incrementalism somehow manage to track state in the `obj` directory as well as in-memory? cc @pranavkm

**Comment by SteveSandersonMS** [comment]
Not surprisingly, adding `<UseRazorSourceGenerator>false</UseRazorSourceGenerator>` makes the problem go away entirely.

**Comment by pranavkm** [comment]
This was tracked down as a compiler bug that is being addressed as part of https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1395568. For the time being the work around is to turn off some caching in the compiler:

```xml
<PropertyGroup>
    <Features>disable-generator-cache</Features>
</PropertyGroup>
```
